src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -827,20 +826,16 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                 contentHeaders = request.Content.Headers;
             }
 
-            string[] allHeaders = HeaderUtilities.DumpHeaders(request.Headers, contentHeaders)
-                .Split(s_headerDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            // Prepare request headers
+            List<string> allHeaders = PrepareRequestHeaders(request.Headers, contentHeaders);
```

Comment:
Updated the PR by incorporating comments. There can be duplicate headers provided by developers. But when the flow reaches here (at above code point), the headers are merged. For e.g. in case of below code
HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Put, "http://url");
request.Headers.Add("User-Agent", "B1");
request.Headers.Add("User-Agent", "B2");

The request.header will contain only one header named User-Agent with 2 values B1 & B2.
The out request to curl/winHttp will only contain one header with value as B1 & B2.
This is the behaviour in both winhttp & currently xplat (after this fix). In fact earlier Xplat was sending two out headers
User-Agent: B1
User-Agent: B2
causing the issue Brennan was facing. Moreover now I am using same inbuilt function to get the header values string which is used in winHttpHandler sendRequest.